.github/workflows/unit-tests.yml

```diff
@@ -45,205 +272,99 @@ jobs:
+          php .github/build-packages.php HEAD^ %SYMFONY_VERSION% src\Symfony\Bridge\PhpUnit
+          SET COMPOSER_ROOT_VERSION=%SYMFONY_VERSION%.x-dev
 
-      - name: Install dependencies
-        run: |
-          echo "::group::composer update"
-          $COMPOSER_UP
-          echo "::endgroup::"
+          echo ::group::composer update
+          php composer.phar update --no-progress --ansi
+          echo ::endgroup::
 
-          echo "::group::install phpunit"
-          ./phpunit install
-          echo "::endgroup::"
+          echo ::group::phpunit install
+          php phpunit install
+          echo ::endgroup::
 
-      - name: Patch return types
-        if: "matrix.php == '8.2' && ! matrix.mode"
-        run: |
-          patch -sp1 < .github/expected-missing-return-types.diff
-          git add .
-          composer install -q --optimize-autoloader || composer install --optimize-autoloader
-          SYMFONY_PATCH_TYPE_DECLARATIONS='force=2&php=8.2' php .github/patch-types.php
-          git checkout src/Symfony/Contracts/Service/ResetInterface.php
-          SYMFONY_PATCH_TYPE_DECLARATIONS='force=2&php=8.2' php .github/patch-types.php # ensure the script is idempotent
-          git checkout src/Symfony/Contracts/Service/ResetInterface.php
-          git diff --exit-code
-
-      - name: Check return types
-        if: "matrix.php == '8.2' && ! matrix.mode"
-        run: |
-          php .github/patch-types.php lint
+          echo ::group::memurai-developer install
+          choco install --no-progress memurai-developer
+          echo ::endgroup::
 
       - name: Run tests
         run: |
-          _run_tests() {
-            local ok=0
-            local title="$1$FLIP"
-            local start=$(date -u +%s)
-            OUTPUT=$(bash -xc "$2" 2>&1) || ok=$?
-            local end=$(date -u +%s)
-
-            if [[ $ok -ne 0 ]]; then
-              printf "\n%-70s%10s\n" $title $(($end-$start))s
-              echo "$OUTPUT"
-              echo "Job exited with: $ok"
-              echo -e "\n::error::KO $title\\n"
-            else
-              printf "::group::%-68s%10s\n" $title $(($end-$start))s
-              echo "$OUTPUT"
-              echo -e "\n\\e[32mOK\\e[0m $title\\n\\n::endgroup::"
-            fi
-
-            [[ "${{ matrix.mode }}" = experimental ]] || (exit $ok)
+          $env:Path = 'c:\php;' + $env:Path
+          $env:SYMFONY_PHPUNIT_SKIPPED_TESTS = 'phpunit.skipped'
+
+          # minimal extensions
+          Copy c:\php\php.ini-min c:\php\php.ini
+          mv src\Symfony\Component\HttpClient\phpunit.xml.dist src\Symfony\Component\HttpClient\phpunit.xml
+          php phpunit src\Symfony --exclude-group tty --exclude-group benchmark --exclude-group intl-data --exclude-group network --exclude-group transient-on-windows
+          php phpunit src\Symfony\Component\HttpClient
+
+          # all extensions
```

Comment:
Running tests with minimal extensions and with all extensions should probably be separate steps in the job, to provide better feedback about what is failing for contributors seeing the workflow output.
We can still guarantee that both kind of tests are run by adding an id (let's say `end_of_setup`) on the last step corresponding to the setup of the job, and then using `if: always() && steps.end_of_setup.outcome == 'success'` on the testing step so that running or skipping them depends on the outcome of the setup rather than on the outcome of the previous step (that's what `always()` is about: disabling the condition checking the outcome of the previous step)